File: python/pe4.py
# Largest Palindrome Product

print("=" * 15, "BILANGAN PALINDROM", "=" * 15)
print("=" * 50)
print("Bilangan palindrom adalah bilangan yang jka dibaca \ndari kiri ataupun dari kanan menghasilkan bilangan \nyang sama")
input_digit = int(input("Masukkan digit bilangan")) 


def cek_palindrom(bil):
    str_num = str(bil)
    panjang = len(str_num)
    for x in range(panjang//2):
        if str_num[x] == str_num[panjang - 1 - x]:
            cek = True
        else:
            cek = False
            break
    return cek
    
input_ = input_digit
max_digit  = 10 ** input_ - 1
min_digit = 10 ** (input_ - 1)
list_palindrom = []

for i in range(max_digit, min_digit, -1):
    for x in range(i, min_digit, -1):
        a = i * x
        cek_palindrom(a)
        if cek_palindrom(a) == True:
            list_palindrom.append(a)
        else:
            pass
list_palindrom.sort()
result = list_palindrom[-1]
print("Bilangan palindrom terbesar yang dapat dibentuk \ndari", input_digit, "digit, adalah", result)



        

File: javascript/pe4.js
// Largest Palindrome Product
const digit = 3;

const minRes = 10 ** digit;

function checkPalindrome(num) {
  strNum = num.toString();
  let check = false;
  for (let i = 0; i < parseInt(strNum.length / 2); i++) {
    if (strNum[i] === strNum[strNum.length - 1 - i]) {
      check = true;
    } else {
      check = false;
      break;
    }
  }
  return check;
}

const temp = [];
for (let i = 10 ** (digit - 1); i < minRes; i++) {
  for (let j = i; j < minRes; j++) {
    temp.push(i * j);
  }
}

const palindrome = [];
temp.forEach((element) => {
  if (checkPalindrome(element)) {
    palindrome.push(element);
  }
});

console.log(
  palindrome.sort(function (a, b) {
    return b - a;
  })
);
